File: ruby/Composite/composite_pattern.rb
require_relative 'block'
require_relative 'group'
require_relative 'structure'

block1 = Block.new(2,3);
block2 = Block.new(1,4);
block3 = Block.new(3,7);

# Initialize three structure
structure = Structure.new;
structure1 = Structure.new;
structure2 = Structure.new;

# Composes the groups
structure1.add(block1);
structure1.add(block2);

structure2.add(block3);

structure.add(structure1);
structure.add(structure2);

structure.assemble;

File: java/Composite/src/CompositePattern.java
public class CompositePattern {
	 public static void main(String[] args) {
	        //Initialize three blocks
		 
		 Block block1 = new Block(2,3);
	        Block block2 = new Block(1,4);
	        Block block3 = new Block(3,7);

	        //Initialize three structure
	        Structure structure = new Structure();
	        Structure structure1 = new Structure();
	        Structure structure2 = new Structure();

	        //Composes the groups
	        structure1.add(block1);
	        structure1.add(block2);

	        structure2.add(block3);

	        structure.add(structure1);
	        structure.add(structure2);

	        structure.assemble();
	    }
}